plugin/src/main/java/org/wso2/carbon/uuf/maven/parser/DependencyTreeParser.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.uuf.maven.parser;
+
+import org.wso2.carbon.uuf.maven.exception.ParsingException;
+import org.wso2.carbon.uuf.maven.model.DependencyNode;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+/**
+ * Parser for dependency tree generated by executing goal {@code tree} of Maven Dependency plugin with {@code
+ * outputType=text} and {@code verbose=true} parameters.({@code mvn dependency:tree -DoutputType=text -Dverbose=true})
+ *
+ * @since 1.0.0
+ */
+public class DependencyTreeParser {
+
+    /**
+     * Parses the specified dependency tree file.
+     *
+     * @param dependencyTreeFilePath path to dependency tree file
+     * @return root node of the dependency tree
+     * @throws ParsingException if cannot read or parse the content of the dependency tree file
+     */
+    public DependencyNode parse(String dependencyTreeFilePath) throws ParsingException {
+        List<String> lines;
+        try {
+            lines = Files.readAllLines(Paths.get(dependencyTreeFilePath));
+        } catch (IOException e) {
+            throw new ParsingException("Cannot read the content of dependency tree '" + dependencyTreeFilePath + "'.",
+                                       e);
+        }
+        return parse(lines);
+    }
+
+    /**
+     * Parses the specified dependency tree text.
+     *
+     * @param dependencyTreeLines lines of the dependency tree to parse
+     * @return root node of the dependency tree
+     * @throws ParsingException if an error occurred when parsing the dependency tree
+     */
+    DependencyNode parse(List<String> dependencyTreeLines) throws ParsingException {
```

Comment:
As suggested by @afkham during the review, can you check how the dependencies can be retrieved using maven api.